### DISCUSSION TITLE
nanoCLR win32 release build doesn't compile following the instruction

### DISCUSSION THREAD

**Discussion by TheCandyCircuit** [nanoframework/Home#1191] [General]
Hi all, 
Enitalp here talking for TheCandyCircuit (you will hear from us more when it is time ;p) :
So "fresh pc."
Get https://github.com/nanoframework/nf-interpreter
go there https://github.com/nanoframework/nf-interpreter/tree/main/targets/win32
follow instructions, 
build. 
And it failed because 
VERSION_MAJOR, VERSION_MINOR, VERSION_BUILD
are not defined
Because on a regular PC, NBGV_VersionMajor, etc... are not defined. 

Nothing major ;p 

Should the release version backup to a default version number? and log a warning in the console (one time ;p )
The debug version already does that in some way. It could use the same default fail behavior.
So nothing is defined in the project in debug, release, VERSION_MAJOR = NBGV_VersionMajor
And in a .h, include somewhere one time in the code, 
#ifndef VERSION_MAJOR 
#define VERSION_MAJOR 0
#ifdef DEBUG
#pragma msg('WARNING : You pc is about to explode"):
#endif 

That would make the build more universal, debug config, have less to set up, and release doesn't change. 
I Will join the community Discord.

**Comment by josesimoes** [comment]
Hi @TheCandyCircuit 

A brief explanation: the WIN target it's used in internal tools and development. That's why you see the preprocessor defines being defined on debug builds. For release, which we're expecting to be run on a pipeline those are set in the process. There are no "defaults" for release exactly on purpose so the build breaks in case those don't get defined for some reason.
Adding those default values would defeat this "sanity check" therefore we're not planning to add them. 
But you can, of course, set them locally to fit your purpose.

Out of curiosity: what's the use you're planning for the WIN target?

**Reply by TheCandyCircuit** [reply]
Hi, I understand your points. It was just an idea. 
It's an interesting question ;p.
I'm aiming for HMI "embedded" applications in various domains as examples of a Hardware + Software solution.
Including (here we go) a Windows (first ;p) application (written in C#, and possibly also in NanoFramework) allowing you to edit your application + UI editor, animation. With the ability to press "Play," spawn an instance of NanoFramework with my code loading the UI to test from another instance of NanoFramework.  
To resume it fast ;p 
Oh yes, open source, of course.
PS: It's not my first rodeo. We can discuss it more if you want on discord.
